cmd/backup-manager/app/backup/backup.go

```diff
@@ -14,200 +14,128 @@
 	}
-	return nil
+	glog.Infof("Backup data for cluster %s successfully, output: %s", bo, string(output))
+	return path, nil
 }
 
-func (bo *BackupOpts) dumpTidbClusterData() (string, error) {
-	bfPath := bo.getBackupFullPath()
-	err := util.EnsureDirectoryExist(bfPath)
+// getCommitTs get backup position from `EndVersion` in BR backup meta
+func getCommitTs(backup *v1alpha1.Backup) (uint64, error) {
+	var commitTs uint64
+	s, err := util.NewRemoteStorage(backup)
 	if err != nil {
-		return "", err
-	}
-	args := []string{
-		fmt.Sprintf("--outputdir=%s", bfPath),
-		fmt.Sprintf("--host=%s", bo.Host),
-		fmt.Sprintf("--port=%d", bo.Port),
-		fmt.Sprintf("--user=%s", bo.User),
-		fmt.Sprintf("--password=%s", bo.Password),
-		"--long-query-guard=3600",
-		"--tidb-force-priority=LOW_PRIORITY",
-		"--verbose=3",
-		"--regex",
-		"^(?!(mysql|test|INFORMATION_SCHEMA|PERFORMANCE_SCHEMA))",
+		return commitTs, err
 	}
-
-	output, err := exec.Command("/mydumper", args...).CombinedOutput()
+	defer s.Close()
+	ctx := context.Background()
+	exist, err := s.Exists(ctx, MetaFile)
 	if err != nil {
-		return bfPath, fmt.Errorf("cluster %s, execute mydumper command %v failed, output: %s, err: %v", bo, args, string(output), err)
+		return commitTs, err
 	}
-	return bfPath, nil
-}
+	if !exist {
+		return commitTs, fmt.Errorf("%s not exist", MetaFile)
 
-func (bo *BackupOpts) backupDataToRemote(source, bucketURI string) error {
-	destBucket := util.NormalizeBucketURI(bucketURI)
-	tmpDestBucket := fmt.Sprintf("%s.tmp", destBucket)
-	// TODO: We may need to use exec.CommandContext to control timeouts.
-	output, err := exec.Command("rclone", constants.RcloneConfigArg, "copyto", source, tmpDestBucket).CombinedOutput()
-	if err != nil {
-		return fmt.Errorf("cluster %s, execute rclone copyto command for upload backup data %s failed, output: %s, err: %v", bo, bucketURI, string(output), err)
 	}
-
-	glog.Infof("upload cluster %s backup data to %s successfully, now move it to permanent URL %s", bo, tmpDestBucket, destBucket)
-
-	// the backup was a success
-	// remove .tmp extension
-	output, err = exec.Command("rclone", constants.RcloneConfigArg, "moveto", tmpDestBucket, destBucket).CombinedOutput()
+	metaData, err := s.ReadAll(ctx, MetaFile)
 	if err != nil {
-		return fmt.Errorf("cluster %s, execute rclone moveto command failed, output: %s, err: %v", bo, string(output), err)
+		return commitTs, err
 	}
-	return nil
-}
-
-func (bo *BackupOpts) cleanRemoteBackupData(bucket string) error {
-	destBucket := util.NormalizeBucketURI(bucket)
-	output, err := exec.Command("rclone", constants.RcloneConfigArg, "deletefile", destBucket).CombinedOutput()
+	backupMeta := &kvbackup.BackupMeta{}
+	err = proto.Unmarshal(metaData, backupMeta)
 	if err != nil {
-		return fmt.Errorf("cluster %s, execute rclone deletefile command failed, output: %s, err: %v", bo, string(output), err)
+		return commitTs, err
```

Comment:
ditto

Reply:
Here is to return error, the caller will log the error with backup CR info.